c_tests/run.rs

```diff
@@ -103,6 +103,8 @@ fn mk_compiler(exe: &Path, src: &Path, opt: &str, extra_objs: &[PathBuf]) -> Com
         "-Wall",
         // Enable LTO with lld.
         "-fuse-ld=lld",
+        // Don't remove frame pointers as we'll need them for guard failures.
+        "-fno-omit-frame-pointer",
```

Comment:
So we couldn't address from the stack pointer? I guess it complicates things a lot?

Reply:
If I remember correctly this is needed so we can get the return address with `__builtin_return_address(0)`.

Reply:
Before we turn frame pointers off (which, AIUI, can have a big effect on optimisations), we should see if there's another way of doing this.

Reply:
Looks like there's a `RA` register on x64 which supposedly contains the return address. Let me see if that works.

Reply:
That's the first I've heard of this. Got a link?

Reply:
Ah, so `fno-omit-frame-pointer` stores the return address in the `RA` register. With `fomit-frame-pointer` this register can be used as general purpose register, which explains why this may have a performance implication. So we may have to read the return address from the stack manually.

Reply:
Fixed in 4cd89ad